src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
-
-        for (int i = 0; i < new InvWrapper(worker.getInventoryCitizen()).getSlots(); i++)
+        setDelay(20);
+        for (final ToolType toolType : toolsNeeded)
         {
-            final ItemStack stack = worker.getInventoryCitizen().getStackInSlot(i);
-
-            if (ItemStackUtils.isEmpty(stack))
+            if (getOwnBuilding() != null && !InventoryUtils.hasItemHandlerToolWithLevel(new InvWrapper(getInventory()),
+              toolType,
+              0,
+              getOwnBuilding().getMaxToolLevel()))
             {
-                new InvWrapper(worker.getInventoryCitizen()).extractItem(i, Integer.MAX_VALUE, false);
-                continue;
+                return START_WORKING;
             }
+        }
 
-            if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == ItemStackUtils.EMPTY)
+        if (worker.getWorkBuilding() != null
+              && !(worker.getLastAttackedEntity() != null
+              && !worker.getLastAttackedEntity().isDead)
+              && getOwnBuilding() instanceof AbstractBuildingGuards
+              && target == null)
+        {
+            final AbstractBuildingGuards guardBuilding = (AbstractBuildingGuards) getOwnBuilding();
+
+            switch (guardBuilding.getTask())
             {
-                worker.setItemStackToSlot(((ItemArmor) stack.getItem()).armorType, stack);
+                case PATROL:
+                    if (currentPatrolPoint == null)
+                    {
+                        currentPatrolPoint = guardBuilding.getNextPatrolTarget(currentPatrolPoint);
+                    }
+                    if (currentPatrolPoint != null
+                      && worker.isWorkerAtSiteWithMove(currentPatrolPoint, 1))
+                    {
+                        currentPatrolPoint = guardBuilding.getNextPatrolTarget(currentPatrolPoint);
+                    }
+                    break;
+                case GUARD:
+                    worker.isWorkerAtSiteWithMove(guardBuilding.getGuardPos(), GUARD_POS_RANGE);
+                    break;
+                case FOLLOW:
+                    worker.isWorkerAtSiteWithMove(guardBuilding.getPlayerToFollow(), GUARD_POS_RANGE);
+                    break;
+                default:
+                    worker.isWorkerAtSiteWithMove(worker.getWorkBuilding().getLocation(), 10);
+                    break;
             }
         }
+
+        if (target == null)
+        {
+            target = getTarget();
+        }
+
+        if (target != null && target.isDead)
+        {
+            worker.addExperience(EXP_PER_MOD_DEATH);
+            target = null;
+        }
+
+        return DECIDE;
     }
 
     /**
-     * Chooses a target from the list.
+     * Get a target for the guard.
      *
-     * @return the next state.
+     * @return The next AIState to go to.
      */
-    protected AIState getTarget()
+    protected EntityLivingBase getTarget()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 33 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)